freshlybuiltimagejaano/downloader.py

```diff
@@ -0,0 +1,149 @@
+        dir_path = path.dirname(path.realpath(__file__))
+        
+        available_models={
+            "img_desc":["model_29","57.2MB","3f2502cb4bdb483be654db226e7610be"],
+            "img_desc_r":["model_r_","57.2MB","d7c8e7bc24d64981e658771f860229b9"]
+            }
+        
+        if path.isdir(dir_path+"models/")==False:
+            try:
+                mkdir(dir_path+"/models")
+            except:
+                print("models directory found")
+            
+        if model_name in available_models:
+            model_name=available_models[model_name]
+            if path.isfile(dir_path+"/models/"+model_name[0]+".h5")==False:
+                try:
+                    print('starting model download')
+                    print("don't quit until downlaoding completes")
+                    print('download can take time depending upon your internet conection')
+                    print(Fore.BLUE+model_name[0]+" is of "+model_name[1])
+                    choice=input(Fore.YELLOW+"do you wish to download type 'y':")
+                    if (choice=='y'):
+                        return self.start_downloading(model_name,dir_path,status_code)
+                    else:
+                        print('download canceled')
+                        status_code=0000
+                        return status_code
+                    print('model download successful')
+                    self.status_code=1003
+                    return status_code
+                except:
+                    print('download interrupted')
+                    try:
+                        remove(dir_path+"/models/"+model_name[0] +".pb")
+                        self.status_code=1002
+                        return status_code
+                    except:
+                        self.status_code=1002
+                        return status_code
+            else:
+                print('model found')
+                print('checking encryption signature')
+                self.hash_signature_match(model_name,available_models,dir_path,status_code)
+                if self.status_code==1000:
+                    return self.status_code
+                else:
+                    print(self.status_code)
+                    print(Fore.CYAN+model_name[0]+" is of "+model_name[1])
+                    re_choice=input(Fore.YELLOW+"press 'y' to start re-downloading: ")
+                    if re_choice =='y':
+                        remove(dir_path+"/models/"+model_name[0] +".pb")
+                        self.start_downloading(model_name,dir_path,status_code)
+                        self.hash_signature_match(model_name,available_models,dir_path,status_code)
+                        return self.status_code
+                
+        else: 
+            print("no reference found for "+model_name)
+            self.status_code=1001
+            return status_code
+    
+    
+    def start_downloading(self,model_name,dir_path,status_code):
```

Comment:
Please add a docstring to the function and class,

Reply:
done